src/test/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/EntityComponentSpec.scala

```diff
@@ -0,0 +1,437 @@
+
+      withWorkspaceContext(bugData.workspace) { context =>
+        assertResult(Some(bugData.sample1)) {
+          runAndWait(entityQuery.get(context, "Sample", "sample1"))
+        }
+      }
+
+  }
+
+  it should "get an entity" in withDefaultTestDatabase {
+
+      withWorkspaceContext(testData.workspace) { context =>
+        assertResult(Some(testData.pair1)) {
+          runAndWait(entityQuery.get(context, "Pair", "pair1"))
+        }
+        assertResult(Some(testData.sample1)) {
+          runAndWait(entityQuery.get(context, "Sample", "sample1"))
+        }
+        assertResult(Some(testData.sset1)) {
+          runAndWait(entityQuery.get(context, "SampleSet", "sset1"))
+        }
+      }
+
+  }
+
+  it should "return None when an entity does not exist" in withDefaultTestDatabase { 
+    
+      withWorkspaceContext(testData.workspace) { context =>
+        assertResult(None) {
+          runAndWait(entityQuery.get(context, "pair", "fnord"))
+        }
+        assertResult(None) {
+          runAndWait(entityQuery.get(context, "fnord", "pair1"))
+        }
+      }
+
+  }
+
+  it should "save a new entity" in withDefaultTestDatabase { 
+
+      withWorkspaceContext(testData.workspace) { context =>
+        val pair2 = Entity("pair2", "Pair",
+          Map(
+            "case" -> AttributeEntityReference("Sample", "sample3"),
+            "control" -> AttributeEntityReference("Sample", "sample1")))
+        runAndWait(entityQuery.save(context, pair2))
+        assert {
+          runAndWait(entityQuery.get(SlickWorkspaceContext(testData.workspace), "Pair", "pair2")).isDefined
+        }
+      }
+
+  }
+
+  it should "clone all entities from a workspace containing cycles" in withDefaultTestDatabase { 
+    val workspaceOriginal = Workspace(
+      namespace = testData.wsName.namespace + "Original",
+      name = testData.wsName.name + "Original",
+      workspaceId = UUID.randomUUID.toString,
+      bucketName = "aBucket",
+      createdDate = DateTime.now,
+      lastModified = DateTime.now,
+      createdBy = "Joe Biden",
+      Map.empty,
+      Map.empty
+    )
+
+    val workspaceClone = Workspace(
+      namespace = testData.wsName.namespace + "Clone",
+      name = testData.wsName.name + "Clone",
+      workspaceId = UUID.randomUUID.toString,
+      bucketName = "anotherBucket",
+      createdDate = DateTime.now,
+      lastModified = DateTime.now,
+      createdBy = "Joe Biden",
+      Map.empty,
+      Map.empty
+    )
+
+
+      val c1 = Entity("c1", "samples", Map("foo" -> AttributeString("x"), "bar" -> AttributeNumber(3), "cycle1" -> AttributeEntityReference("samples", "c2")))
+      val c2 = Entity("c2", "samples", Map("foo" -> AttributeString("x"), "bar" -> AttributeNumber(3), "cycle2" -> AttributeEntityReference("samples", "c3")))
+      var c3 = Entity("c3", "samples", Map("foo" -> AttributeString("x"), "bar" -> AttributeNumber(3)))
```

Comment:
I know it's not your original code, but I found 4 vars in this file that I want to banish.  Please eliminate or justify.